modules/smmsBidAdapter.js

```diff
@@ -0,0 +1,160 @@
+import * as utils from '../src/utils';
+import { registerBidder } from '../src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'smms';
+const ENDPOINT_BANNER = '//bidder.mediams.mb.softbank.jp/api/v1/prebid/banner';
+const ENDPOINT_NATIVE = '//bidder.mediams.mb.softbank.jp/api/v1/prebid/native';
+const COOKIE_SYNC_URL = '//bidder.mediams.mb.softbank.jp/api/v1/cookie/gen';
+const SUPPORTED_MEDIA_TYPES = ['banner', 'native'];
+const SUPPORTED_CURRENCIES = ['USD', 'JPY'];
+const DEFAULT_CURRENCY = 'JPY';
+const NET_REVENUE = true;
+
+function _encodeURIComponent(a) {
+  let b = window.encodeURIComponent(a);
+  b = b.replace(/'/g, '%27');
+  return b;
+}
+
+export function _getUrlVars(url) {
+  let hash;
+  const myJson = {};
+  const hashes = url.slice(url.indexOf('?') + 1).split('&');
+  for (let i = 0; i < hashes.length; i++) {
+    hash = hashes[i].split('=');
+    myJson[hash[0]] = hash[1];
+  }
+  return myJson;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['smms'], // short code
+  supportedMediaTypes: SUPPORTED_MEDIA_TYPES,
+  isBidRequestValid: function(bid) {
+    let valid = !!(bid.params.placementId);
+    if (valid && bid.params.hasOwnProperty('currency')) {
+      if (SUPPORTED_CURRENCIES.indexOf(bid.params.currency) === -1) {
+        console.log('Invalid currency type, we support only JPY and USD!');
+        valid = false;
+      }
+    }
+
+    return valid;
+  },
+  /**
+    * Make a server request from the list of BidRequests.
+    *
+    * @param {validBidRequests[]} - an array of bids
+    * @return ServerRequest Info describing the request to the server.
+    */
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const serverRequests = [];
+    let refererInfo;
+    if (bidderRequest && bidderRequest.refererInfo) {
+      refererInfo = bidderRequest.refererInfo;
+    }
+    const g = (typeof (geparams) !== 'undefined' && typeof (geparams) == 'object' && geparams) ? geparams : {};
+    validBidRequests.forEach((bid, i) => {
+      let endpoint = ENDPOINT_BANNER;
+      let data = {
+        'placementid': bid.params.placementId,
+        'cur': bid.params.hasOwnProperty('currency') ? bid.params.currency : DEFAULT_CURRENCY,
+        'ua': navigator.userAgent,
+        'adtk': _encodeURIComponent(g.lat ? '0' : '1'),
+        'loc': utils.getTopWindowUrl(),
```

Comment:
This will be removed in 3.0 please use `refererInfo.referer` instead

Reply:
I changed to `refererInfo.referer`.